client/components/data/query-post-counts/index.jsx

```diff
@@ -1,58 +1,33 @@
-/** @format */
-
 /**
  * External dependencies
  */
-
 import PropTypes from 'prop-types';
-import { Component } from 'react';
-import { connect } from 'react-redux';
+import { useEffect } from 'react';
+import { useDispatch } from 'react-redux';
 
 /**
  * Internal dependencies
  */
 import { requestPostCounts } from 'state/posts/counts/actions';
 import { isRequestingPostCounts } from 'state/posts/counts/selectors';
 
-class QueryPostCounts extends Component {
-	UNSAFE_componentWillMount() {
-		this.request( this.props );
-	}
-
-	UNSAFE_componentWillReceiveProps( nextProps ) {
-		if ( this.props.siteId === nextProps.siteId && this.props.type === nextProps.type ) {
-			return;
-		}
-
-		this.request( nextProps );
+const request = ( siteId, type ) => ( dispatch, getState ) => {
+	if ( ! isRequestingPostCounts( getState(), siteId, type ) ) {
+		dispatch( requestPostCounts( siteId, type ) );
 	}
+};
 
-	request( props ) {
-		if ( props.requesting ) {
-			return;
-		}
+export default function QuerySiteDomains( { siteId, type } ) {
```

Comment:
Why is the component called `QuerySiteDomains` instead of `QueryPostCounts`?

Reply:
Because I make silly copy-paste mistakes 😅